src/posit/connect/packages.py

```diff
@@ -1,213 +1,99 @@
-            if e.http_status == 204:
-                return []
-            raise e
-
-    def find(self, uid):
-        raise NotImplementedError("The 'find' method is not support by the Packages API.")
-
-    class _FindBy(TypedDict, total=False):
-        language: NotRequired[Literal["python", "r"]]
-        """Programming language ecosystem, options are 'python' and 'r'"""
-
-        name: NotRequired[str]
-        """The package name"""
-
-        version: NotRequired[str]
-        """The package version"""
-
-        hash: NotRequired[Optional[str]]
-        """Package description hash for R packages."""
-
-    def find_by(self, **conditions: Unpack[_FindBy]):  # type: ignore
+    def find_by(
+        self,
+        *,
+        language: Literal["python", "r"] = ...,
+        name: str = ...,
+        version: str = ...,
+        hash: str | None = ...,  # noqa: A002
+    ) -> _ContentPackage | None:
         """
         Find the first record matching the specified conditions.
 
         There is no implied ordering, so if order matters, you should specify it yourself.
 
         Parameters
         ----------
-        **conditions : Unpack[_FindBy]
-            Conditions for filtering packages. The following keys are accepted:
-
         language : {"python", "r"}, not required
             Programming language ecosystem, options are 'python' and 'r'
-
         name : str, not required
             The package name
-
         version : str, not required
             The package version
-
         hash : str or None, optional, not required
             Package description hash for R packages.
 
         Returns
         -------
-        Optional[T]
+        _ContentPackage | None
             The first record matching the specified conditions, or `None` if no such record exists.
         """
-        return super().find_by(**conditions)
-
-
-class ContentPackagesMixin(Active):
-    """Mixin class to add a packages attribute."""
-
-    @property
-    @requires(version="2024.10.0-dev")
-    def packages(self):
-        path = posixpath.join(self._path, "packages")
-        return ContentPackages(self._ctx, path)
-
-
-class Package(Active):
-    class _Package(TypedDict):
-        language: Required[Literal["python", "r"]]
-        """Programming language ecosystem, options are 'python' and 'r'"""
-
-        language_version: Required[str]
-        """Programming language version"""
-
-        name: Required[str]
-        """The package name"""
+        ...
 
-        version: Required[str]
-        """The package version"""
 
-        hash: Required[Optional[str]]
-        """Package description hash for R packages."""
+class _Package(Mapping[str, Any]):
+    pass
 
-        bundle_id: Required[str]
-        """The unique identifier of the bundle this package is associated with"""
 
-        app_id: Required[str]
-        """The numerical identifier of the application this package is associated with"""
+class _Packages(Sized, Protocol):
```

Comment:
Is it possible to suffix the class names with `P` to help visually declare that it's a protocol class?

```suggestion
class _PackagesP(Sized, Protocol):
```

(Much easier to perform with "rename symbol")

Reply:
Actually, I think we should make these public types. External users may want to use them for type hinting.

For example...

```py
from posit.connect import Client, Packages

client = Client()
packages: Packages = client.packages
```